src/ophyd_async/tango/core/_tango_transport.py

```diff
@@ -615,6 +603,80 @@ def get_trl_descriptor(
     raise RuntimeError(f"Error getting descriptor for {tango_resource}")
 
 
+def get_trl_descriptor(
+    datatype: type | None,
+    tango_resource: str,
+    tr_configs: dict[str, AttributeInfoEx | CommandInfo],
+) -> DataKey:
+    descriptor = get_simple_datakey(datatype, tango_resource, tr_configs)
+
+    for _, config in tr_configs.items():
+        if isinstance(config, AttributeInfoEx):
+            alarm_info = config.alarms
+            _limits = Limits(
+                control=LimitsRange(
+                    low=try_to_cast_as_float(config.min_value),
+                    high=try_to_cast_as_float(config.max_value),
+                ),
+                warning=LimitsRange(
+                    low=try_to_cast_as_float(alarm_info.min_warning),
+                    high=try_to_cast_as_float(alarm_info.max_warning),
+                ),
+                alarm=LimitsRange(
+                    low=try_to_cast_as_float(alarm_info.min_alarm),
+                    high=try_to_cast_as_float(alarm_info.max_alarm),
+                ),
+            )
+
+            delta_t, delta_val = map(
+                try_to_cast_as_float, (alarm_info.delta_t, alarm_info.delta_val)
+            )
+            limits_rds = RdsRange(
+                time_difference=delta_t or 0.0,
+                value_difference=delta_val or 0.0,
+            )
+
+            if any((delta_t, delta_val)):
+                _limits["rds"] = limits_rds
```

Comment:
Could `delta_t`, `delta_val` ever intentionally be `(0.0, 0.0)` instead of `(None, None)`? If so this check would leave this data out of the datakey.

```suggestion
            if (alarm_info.delta_t, alarm_info.delta_val) != (None, None):
                delta_t = try_to_cast_as_float(alarm_info.delta_t) or 0.0
                delta_val = try_to_cast_as_float(alarm_info.delta_val) or 0.0
                _limits["rds"] = RdsRange(
                    time_difference=delta_t,
                    value_difference=delta_val
                )
```